src/H5Gloc.c

```diff
@@ -927,13 +926,15 @@ H5G__loc_set_comment_cb(H5G_loc_t H5_ATTR_UNUSED *grp_loc /*in*/, const char H5_
 
     /* Add the new message */
     if (udata->comment && *udata->comment) {
-        /* Casting away const OK -QAK */
-        comment.s = (char *)udata->comment;
+        if (NULL == (comment.s = HDstrdup(udata->comment)))
```

Comment:
Short of playing games with our flexible const pointer type, the easiest way to solve this appears to be taking a copy of the object comment passed in from above, since H5O_msg_create accepts a non-const message pointer.